src/App.js

```diff
@@ -1,24 +1,406 @@
+        let plusZeros = 0;
+        const tempArr = [];
+
+        for (let i = 0; i < 4; i += 1) {
+          tempArr.push([...prevState.mainGameField[i]]);
+          tempArr[i].sort((a, b) => {
+            if (a === 0 && b !== 0) {
+              return -1;
+            }
+
+            if (a !== 0 && b !== 0) {
+              return 0;
+            }
+
+            if (a !== 0 && b === 0) {
+              return 1;
+            }
+
+            return 0;
+          });
+
+          for (let i2 = 3; i2 > 0; i2 -= 1) {
+            if (tempArr[i][i2] === tempArr[i][i2 - 1]
+              && tempArr[i][i2] !== 0) {
+              tempArr[i][i2] *= 2;
+              if (tempArr[i][i2] === 2048) {
+                this.playerWins();
+              }
+
+              tempArr[i].splice(i2 - 1, 1);
+              tempArr[i].unshift(0);
+              plusZeros += 1;
+            }
+          }
+        }
+
+        return ({
+          mainGameField: [...tempArr],
+          zeros: prevState.zeros + plusZeros,
+        });
+      });
+      setTimeout(this.addNewField, 500);
+    }
+
+    if (event.key === 'ArrowLeft') {
+      this.setState((prevState) => {
+        let plusZeros = 0;
+        const tempArr = [];
+
+        for (let i = 0; i < 4; i += 1) {
+          tempArr.push([...prevState.mainGameField[i]]);
+          tempArr[i].sort((a, b) => {
+            if (a === 0 && b !== 0) {
+              return 1;
+            }
+
+            if (a !== 0 && b !== 0) {
+              return 0;
+            }
+
+            if (a !== 0 && b === 0) {
+              return -1;
+            }
+
+            return 0;
+          });
+
+          for (let i2 = 0; i2 < 3; i2 += 1) {
+            if (tempArr[i][i2] === tempArr[i][i2 + 1]
+              && tempArr[i][i2] !== 0) {
+              tempArr[i][i2] *= 2;
+              if (tempArr[i][i2] === 2048) {
+                this.playerWins();
+              }
+
+              tempArr[i].splice(i2 + 1, 1);
+              tempArr[i].push(0);
+              plusZeros += 1;
+            }
+          }
+        }
+
+        return ({
+          mainGameField: [...tempArr],
+          zeros: prevState.zeros + plusZeros,
+        });
+      });
+      setTimeout(this.addNewField, 500);
+    }
+
+    if (event.key === 'ArrowDown') {
+      this.setState((prevState) => {
+        let plusZeros = 0;
+        const tempArr = [...prevState.mainGameField];
+
+        for (let i = 0; i < 4; i += 1) {
+          const tempColumnArr = [];
+
+          for (let i3 = 0; i3 < 4; i3 += 1) {
+            tempColumnArr.push(prevState.mainGameField[i3][i]);
+          }
+
+          tempColumnArr.sort((a, b) => {
+            if (a === 0 && b !== 0) {
+              return -1;
+            }
+
+            if (a !== 0 && b !== 0) {
+              return 0;
+            }
+
+            if (a !== 0 && b === 0) {
+              return 1;
+            }
+
+            return 0;
+          });
```

Comment:
you have the same part for two cases so it would be good to add separate function for this and use it in both places